sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1214,6 +1215,13 @@ class CastSuite extends CastSuiteBase {
       checkEvaluation(Cast(Literal(134.12), DecimalType(3, 2)), null)
     }
   }
+
+  test("SPARK-30008: collect_list/collect_set can cast to ArrayType not containsNull") {
+    val list = CollectList(Literal(1))
+    assert(Cast.canCast(list.dataType, ArrayType(IntegerType, false)))
```

Comment:
Shall we check the negative case `ArrayType(IntegerType, true)`, too?

Reply:
we set to src array type(`containsNull = false`), then it should be able to convert to the dist both the dist array type is `containsNull` or not. So the case you list here is not negative and it is not a bug. thanks.